Nanoserver is missing DLLs required

**What steps did you take and what happened:**
Set up k8s cluster in azure and ran sonobuoy; ran into 

```
sonobuoy logs
namespace="sonobuoy" pod="sonobuoy" container="kube-sonobuoy"
panic: Failed to load netapi32.dll: The specified module could not be found.

goroutine 1 [running]:
syscall.(*LazyProc).mustFind(...)
	/usr/local/go/src/syscall/dll_windows.go:322
syscall.(*LazyProc).Addr(...)
	/usr/local/go/src/syscall/dll_windows.go:329
syscall.NetGetJoinInformation(0x0, 0xc00011d238, 0xc00011d234, 0x2, 0xc00003a800)
	/usr/local/go/src/syscall/zsyscall_windows.go:1177 +0x105
os/user.isDomainJoined(0xc00011d290, 0xfa272e, 0x0)
	/usr/local/go/src/os/user/lookup_windows.go:18 +0x5a
os/user.lookupFullName(0xc000035220, 0xc, 0xc000035210, 0xd, 0xc00003a800, 0x1a, 0xd, 0xc00003a800, 0x1a, 0x1)
	/usr/local/go/src/os/user/lookup_windows.go:51 +0x2d
os/user.newUser(0xc0000351e0, 0xc, 0xc000035200, 0xc, 0xc00003a7e0, 0x16, 0xc000035210, 0xd, 0xc000035220, 0xc, ...)
	/usr/local/go/src/os/user/lookup_windows.go:181 +0xd4
os/user.current(0x0, 0x0, 0x0)
	/usr/local/go/src/os/user/lookup_windows.go:225 +0x2de
os/user.Current.func1()
	/usr/local/go/src/os/user/lookup.go:15 +0x29
sync.(*Once).doSlow(0x2c548c0, 0x2274318)
	/usr/local/go/src/sync/once.go:68 +0xf7
sync.(*Once).Do(...)
	/usr/local/go/src/sync/once.go:59
os/user.Current(0xc0000351d0, 0x8, 0x21c3b4d)
	/usr/local/go/src/os/user/lookup.go:15 +0x105
k8s.io/klog.init.1()
	/go/pkg/mod/k8s.io/klog@v1.0.0/klog_file.go:58 +0x4b
```

a missing dll and tracked the issue to an upstream one already being discussed: https://github.com/microsoft/Windows-Containers/issues/72

**What did you expect to happen:**
It to run

**Anything else you would like to add:**
Experimenting with testing to verify images before pushing them. Seeing if a golang nanoserver image would work or not or if we have to go to servercore (much bigger)

--- a/DockerfileWindows
+++ b/DockerfileWindows
@@ -12,8 +12,10 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-ARG BASEIMAGE
-FROM ${BASEIMAGE}
+ARG VERSION
+FROM mcr.microsoft.com/windows/servercore:$VERSION as core
+FROM mcr.microsoft.com/windows/nanoserver:$VERSION
+COPY --from=core /Windows/System32/netapi32.dll /Windows/System32/netapi32.dll
 
 ADD build/windows/amd64/sonobuoy.exe /sonobuoy.exe
 WORKDIR /

--- a/scripts/build_funcs.sh
+++ b/scripts/build_funcs.sh
@@ -85,13 +85,12 @@ build_container_dockerfile_arch() {
 }
 
 buildx_container_windows_version(){
-    BASEIMAGE="$WIN_AMD64_BASEIMAGE:$1"
     mkdir -p "build/windows/$WIN_ARCH/$VERSION"
     docker buildx build --pull \
         --output=type=oci,dest=build/windows/$WIN_ARCH/$VERSION/sonobuoy-img-win-$WIN_ARCH-$VERSION-$GITHUB_RUN_ID.tar \
         --platform windows/amd64 \
         -t $REGISTRY/$TARGET:win-$WIN_ARCH-$VERSION-$IMAGE_VERSION \
-        --build-arg BASEIMAGE=$BASEIMAGE \
+        --build-arg VERSION=$1 \
         -f build/windows/$WIN_ARCH/Dockerfile \
         .
 }

